napari/layers/points/points.py

```diff
@@ -297,6 +311,17 @@ def __init__(
             }
             self._properties = empty_properties
 
+        # make the text
+        if type(text) in (list, np.ndarray, str) or text is None:
```

Comment:
```suggestion
        if text is None or isinstance(text, [list, np.ndarray, str]):
```

(I've changed the order because `is None` will be by far the most common branch point and we might as well shortcut it!)